crates/futures/src/lib.rs

```diff
@@ -110,14 +110,35 @@ pub mod futures_0_3;
 use std::cell::{Cell, RefCell};
 use std::fmt;
 use std::rc::Rc;
+#[cfg(target_feature = "atomics")]
+use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
+#[cfg(target_feature = "atomics")]
+use std::sync::Mutex;
 
 use futures::executor::{self, Notify, Spawn};
 use futures::future;
 use futures::prelude::*;
 use futures::sync::oneshot;
 use js_sys::{Function, Promise};
+#[cfg(target_feature = "atomics")]
+use js_sys::{Atomics, Int32Array, SharedArrayBuffer, WebAssembly};
 use wasm_bindgen::prelude::*;
+#[cfg(target_feature = "atomics")]
```

Comment:
Organizationally could there perhaps be:

```
src/stable.rs // what lib.rs currently is
src/atomics.rs // what this file is currently
src/futures03.rs // implementation with futures 0.3
```

and eventually we can add `src/futures03_atomics.rs`, but for now we'd just go with what's here. I don't think we'll benefit too too much from shared types, but the shared pieces can always be in `src/lib.rs`

Reply:
I moved current implementation to the `atomics.rs` file. Do I need to go farther and move contents of `lib.rs` to `stable.rs`?